tips/TIP-0020/tip-0020.md

```diff
@@ -0,0 +1,497 @@
+
+Example:
+Consider a <i>Transaction Essence</i> containing the <i>UTXO Inputs</i> 0, 1 and 2, where 0 and 2 are both spending outputs belonging to the same Ed25519 address A and 1 is spending from a different address B. This results in the following structure of the <i>Unlocks</i> part:
+| Index | Unlock                                                                             |
+| ----- | ---------------------------------------------------------------------------------------- |
+| 0     | A _Signature Unlock_ holding the Ed25519 signature for address A.                  |
+| 1     | A _Signature Unlock_ holding the Ed25519 signature for address B.                  |
+| 2     | A _Reference Unlock_ which references 0, as both require the same signature for A. |
+
+## Validation
+
+A <i>Transaction Payload</i> has different validation stages, since some validation steps can only be executed when certain information has (or has not) been received. We therefore distinguish between syntactic and semantic validation
+
+The different output types and optional output features introduced by [draft TIP-18](https://github.com/iotaledger/tips/pull/38) add additional constraints to the transaction validation rules, but since these are specific to the given outputs and features, they are discussed for each [output type](https://github.com/lzpap/protocol-rfcs/blob/master/tips/TIP-0018/tip-0018.md#output-design) and [feature type](https://github.com/lzpap/protocol-rfcs/blob/master/tips/TIP-0018/tip-0018.md#output-features) separately.
+
+### Syntactic validation
+
+Syntactic validation is checked as soon as the transaction data has been received in its entirety. It validates the structure but not the signatures of the transaction. If the transaction does not pass this stage, it must not be broadcast further and can be discarded right away.
+
+The following criteria defines whether a payload passes the syntactical validation:
+
+* Essence:
+  * `Transaction Type` value must denote a _Transaction Essence_.
+  * `Network ID` must match the value of the current network.
+  * Inputs:
+    * `Inputs Count` must be 0 < x ≤ `Max Inputs Count`.
+    * For each input the following must be true:
+      * `Input Type` must denote a _UTXO Input_.
+      * `Transaction Output Index` must be 0 ≤ x < `Max Outputs Count`.
+    * Each pair of `Transaction ID` and `Transaction Output Index` must be unique in the list of inputs.
+  * Outputs:
+    * `Outputs Count` must be 0 < x ≤ `Max Outputs Count`.
+    * For each input the following must be true:
+      * `Output Type` must match one of the values described under [Outputs](#outputs).
+      * The output itself must pass syntactic validation.
+    * The sum of all `Amount` fields must not exceed `Max IOTA Supply`.
+  * Payload (if present):
+    * `Payload Type` must match one of the values described under [Payload](#payload).
+    * `Data fields` must be correctly parsable in the context of the `Payload Type`.
```

Comment:
I don't understand this sentence

Reply:
Clarified in https://github.com/iotaledger/tips/pull/40/commits/d9370b586a61719331d7fd1fccea365ee5c5d137